src/audio/mixer.c

```diff
@@ -140,16 +146,16 @@ static int mixer_params(struct comp_dev *dev)
 	}
 
 	/* calculate period size based on config */
-	md->period_bytes = dev->frames * dev->frame_bytes;
-	if (md->period_bytes == 0) {
+	period_bytes = dev->frames * dev->frame_bytes;
+	if (period_bytes == 0) {
 		trace_mixer_error("mixer_params() error: period_bytes = 0");
 		return -EINVAL;
 	}
 
 	sink = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
 
 	/* set downstream buffer size */
-	ret = buffer_set_size(sink, md->period_bytes * config->periods_sink);
+	ret = buffer_set_size(sink, period_bytes * config->periods_sink);
```

Comment:
@tlauda should this be done in prepare() using the params from the sink?

Reply:
@ranj063 Mixer doesn't change the format, so it makes no difference.